jetty-core/jetty-quic/jetty-quic-api/src/main/java/org/eclipse/jetty/quic/api/frames/StreamFrame.java

```diff
@@ -0,0 +1,145 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.quic.api.frames;
+
+import java.nio.ByteBuffer;
+
+import org.eclipse.jetty.quic.api.Session;
+import org.eclipse.jetty.quic.api.Stream;
+
+public class StreamFrame extends Frame.WithStreamId
+{
+    public static final long END_STREAM_MASK = 0x01;
+    public static final long LENGTH_MASK = 0x02;
+    public static final long OFFSET_MASK = 0x04;
+
+    private static long toFrameType(boolean hasOffset, boolean hasLength, boolean endStream)
+    {
+        long result = 0x08;
+        if (hasOffset)
+            result |= OFFSET_MASK;
+        if (hasLength)
+            result |= LENGTH_MASK;
+        if (endStream)
+            result |= END_STREAM_MASK;
+        return result;
+    }
+
+    private final long offset;
+    private final ByteBuffer data;
```

Comment:
I'm looking for how retainability is handled for buffers used by the Stream?  I can see that the `Data` objects produced by the stream are retainable, but I can't see how that might be reflected here.   So I would expect at least a javadoc discussing how retainability is handled for these buffers... if not actual explicit retention here.

Reply:
The frame is just a record, it only carries the data that has been parsed, or whatever the application creates.
Retainability is not relevant here, it is only relevant for `Stream.Data`.